### PR TITLE
build: split out host and target library builds fully

### PR DESCRIPTION
**Pull request by compnerd**
Remove this special case handling for building a host library as a target
library.  This is the last piece needed to support cross-compiling lldb.  As a
bonus, it cleans up some of the logic in our special build system.

The way that CMake works with this is that it builds with the concept of the
target being the host.  You can invoke a sub-cmake to cross-compile the library
with the just built compiler.  Unfortunately, we do not do that and are fighting
CMake which breaks horribly with cross-compilation and platforms that are
sufficiently different.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @gottesmm @Rostepher

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by gottesmm** [comment]
I talked with compnerd offline and asked him to create a separate host library called swiftReflection that is just for hosts and a separate library for the host as a target. As a side-effect, we can clean up lldb's importing of swiftReflection.

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-lldb/pull/1017

@swift-ci please test

**Comment by compnerd** [comment]
Please test with following PR:
apple/swift-lldb#1017

@swift-ci please test

File: stdlib/CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,11 @@ if(SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER)
     message(WARNING "Building the swift runtime using the host compiler, and not the just-built clang.")
   endif()
 else()
+  # NOTE(compnerd) save the original compiler for the host swiftReflection that
+  # we build
+  set(_CMAKE_CXX_COMPILER_INITIAL ${CMAKE_CXX_COMPILER})
```
Comment (gottesmm) [inline stdlib/CMakeLists.txt]: Why not just call it CMAKE_CXX_COMPILER_HOST? That would make it clear. Beyond that looks awesome!

Comment (compnerd) [reply stdlib/CMakeLists.txt]: Naming is not solvable :-p

**Comment by compnerd** [comment]
Please test with following PR:
apple/swift-lldb#1017

@swift-ci please test

**Comment by gottesmm** [comment]
Thanks for changing that variable name. LGTM.

**Review by Rostepher — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**